### PR TITLE
Make engine lighter

### PR DESCRIPTION
**Pull request by ganmacs**
**Which issue(s) this PR fixes**: 
Ref https://github.com/fluent/fluentd/issues/2624

**What this PR does / why we need it**: 

Make engine lighter for safe reloading.

**Docs Changes**:

no need

**Release Note**: 

no need

### CODE REVIEW

File: lib/fluent/fluent_log_event_router.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+module Fluent
+  # DO NOT write any logic here
+  class NullFluentLogEventRouter
+    def start; end
+
+    def stop; end
+
+    def graceful_stop; end
+
+    def emit_event; end
```
Comment (repeatedly) [inline lib/fluent/fluent_log_event_router.rb]: No argument. is it intended?

Comment (ganmacs) [reply lib/fluent/fluent_log_event_router.rb]: Good catch!  I missed adding an argument.
https://github.com/fluent/fluentd/pull/2667/commits/0970e78b9f17b5fcccc2524f529ceba573944b90

Comment (repeatedly) [reply lib/fluent/fluent_log_event_router.rb]: should add test for `NullFluentLogEventRouter` methods?

Comment (ganmacs) [reply lib/fluent/fluent_log_event_router.rb]: I added https://github.com/fluent/fluentd/pull/2667/commits/65b7429c03cc7fc9eead11a506338d9859c44f5e

**Pull request merged**